src/transports/WebSocketTransport.ts

```diff
@@ -29,15 +29,18 @@ class WebSocketTransport extends Transport {
   public async sendData(data: JSONRPCRequestData, timeout: number | null = 5000): Promise<any> {
     let prom = this.transportRequestManager.addRequest(data, timeout);
     const notifications = getNotifications(data);
-    this.connection.send(JSON.stringify(this.parseData(data)), (err?: Error) => {
-      if (err) {
-        const jsonError = new JSONRPCError(err.message, ERR_UNKNOWN, err);
-        this.transportRequestManager.settlePendingRequest(notifications, jsonError);
-        this.transportRequestManager.settlePendingRequest(getBatchRequests(data), jsonError);
-        prom = Promise.reject(jsonError);
-      }
+    try {
+      this.connection.send(JSON.stringify(this.parseData(data)));
```

Comment:
I'm not sure this is right in all instances, so we're using isomorphic-ws , which in turn switches between the browser and nodejs. 

The nodejs implementation allows for a whole callback system, for when you've completed sending the data over the wire. 

The browser system doesn't have the same, I take it that maybe send in the browser is blocking, where as in node it's async via callback.

I might be wrong here, but it seems like the browser has a a different notification system for errors than node.  It also seems like this.connection.send isn't an async/awaitable thing 🤔 .

Reply:
I cant find any docs about this callback, other than for socket.io. 

Any chance you'd feel like pushing a commit to fix it?

Reply:
Actually I think this is fine. The docs say oddly that it does both , there's the callback on send for node.js, but then otherwise it throws an error, so it probably accounts for the missing error in the browser, as it doesn't have the callback interface, just on node. so yep ready to sign off on this . 


https://github.com/websockets/ws/blob/master/doc/ws.md#websocketsenddata-options-callback

Reply:
Seems like the browser throws is the same as the send 
<img width="539" alt="image" src="https://user-images.githubusercontent.com/173187/210899696-5e4c4333-e52c-45c2-919b-2b8ecc2ef7cd.png">